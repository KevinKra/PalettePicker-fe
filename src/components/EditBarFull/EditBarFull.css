.edit-block {
  position: absolute;
  z-index: 2;
  width: 100%;
  padding-top: 12px;
  background-color: white;
  transform: translateY(-110%);
  transition: 0.5s cubic-bezier(0.42, 0.63, 0.46, 1.15);
  border-bottom: 6px solid coral;
  box-shadow: -1px -7px 15px 9px rgba(0, 0, 0, 0.75);
}

.edits-form {
  display: flex;
  justify-content: space-evenly;
  margin: 0.5rem;
}

.hue-selection-inputs {
  display: flex;
  flex-direction: column;
}

.hue-selection-inputs > button {
  height: 30px;
  margin: 0.2rem 0;
  background-color: coral;
  color: white;
  border: 1px solid red;
  border-radius: 3px;
}

.radio-styles {
  display: flex;
  flex-direction: column;
  width: 33%;
  border-left: 1px solid black;
  padding: 0 0.5rem;
}

.radio-styles > label {
  display: flex;
  justify-content: space-between;
}

.update-btn {
  background-color: coral;
  color: white;
  border: 1px solid rgb(199, 94, 56);
  border-radius: 4px;
  padding: 0 1rem;
  cursor: pointer;
  transition: 0.07s all ease-in-out;
  box-shadow: inset 0px 0px 0px 0px rgba(0, 0, 0, 0.75);
}

.update-btn:focus {
  outline: none;
}

.update-btn:active {
  outline: none;
  box-shadow: inset -2px 2px 1px 2px rgba(0, 0, 0, 0.75);
}

.update-btn > p {
  transition: 0.07s all ease-in-out;
  transform: translate(0%, 0%);
}
.update-btn:active > p {
  transform: translate(-6%, 6%);
}

h4 {
  margin: 0;
  margin-bottom: 0.2rem;
  font-size: 16px;
  font-family: "Lobster", cursive;
  letter-spacing: 1px;
  font-weight: 100;
  color: #222;
}
label {
  font-size: 12px;
}
