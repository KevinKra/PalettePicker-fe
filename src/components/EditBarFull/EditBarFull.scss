@import "../../_styles/main.scss";

.edit-block {
	position: absolute;
	z-index: 2;
	width: 100%;
	padding-top: 12px;
	background-color: $primary-light;
	transform: translateY(-30vh);
	transition: 0.5s cubic-bezier(0.42, 0.63, 0.46, 1.15);
	border-bottom: 6px solid $primary-color;
	box-shadow: -1px -7px 15px 9px rgba(0, 0, 0, 0.75);
	h4 {
		margin: 0;
		margin-bottom: 0.2rem;
		font-size: 16px;
		font-family: "Lobster", cursive;
		letter-spacing: 1px;
		font-weight: 100;
		color: #222;
	}
	label {
		font-size: 12px;
	}
	footer {
		display: flex;
		justify-content: center;
		border-top: 1px solid $primary-color;
	}
	.edits-form {
		display: flex;
		justify-content: space-evenly;
		margin: 0.5rem;
	}
	.hue-selection-inputs {
		display: flex;
		flex-direction: column;
	}
	.hue-selection-inputs > button {
		@include btn-style-1;
		height: 24px;
		margin: 0.5rem 0;
	}
	.update-btn {
		@include btn-style-1;
		padding: 0 1rem;
		transition: 0.07s all ease-in-out;
	}
	.close-btn {
		@include btn-style-1;
		padding: .3rem 1.5rem;
		margin: .3rem 0;
	}
	.radio-styles {
		display: flex;
		flex-direction: column;
		width: 33%;
		border-left: 1px solid black;
		padding: 0 0.5rem;
	}
	.radio-styles > label {
		display: flex;
		justify-content: space-between;
	}
}
