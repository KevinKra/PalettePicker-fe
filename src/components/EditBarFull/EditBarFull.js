import React from 'react';
import './EditBarFull.scss';

export default function EditBarFull(props) {
	const editBarActive = {
		transform: 'translateY(-1vh)'
	};

	const generateRadios = (format) => {
		const schemeNames = [ 'mono', 'contrast', 'triade', 'tetrade', 'analogic' ];
		const variationNames = [ 'default', 'pastel', 'soft', 'light', 'hard', 'pale' ];
		// Dropped fully-DRY version because it was difficult to read. I like this version _a lot_ more.
		if (format === 'colors') {
			return schemeNames.map((name, i) => {
				return (
					<label htmlFor={name} key={i}>
						{name}
						<input
							type="radio"
							name="colorScheme-selection"
							value={name}
							defaultChecked={props.colorScheme === name ? true : false}
							onClick={(e) => props.updatePaletteFeature(e, 'colorScheme')}
						/>
					</label>
				);
			});
		} else {
			return variationNames.map((name, i) => {
				return (
					<label htmlFor={name} key={i}>
						{name}
						<input
							type="radio"
							name="variation-selection"
							value={name}
							defaultChecked={props.variation === name ? true : false}
							onClick={(e) => props.updatePaletteFeature(e, 'variation')}
						/>
					</label>
				);
			});
		}
	};

	const colorSchemes = generateRadios('colors');
	const colorVariations = generateRadios('variations');

	return (
		<section className="edit-block" style={props.showFullEditBar ? editBarActive : null}>
			<form className="edits-form" onSubmit={(e) => props.updateColors(e, props.colors)}>
				<div className="hue-selection-container">
					<h4>Hue Selection:</h4>
					<div className="hue-selection-inputs">
						<label htmlFor="hue-selection">
							<input
								type="text"
								placeholder="default: random"
								autoComplete="off"
								name="hue-selection"
								value={props.hue}
								onChange={props.updateHue}
							/>
						</label>
						{props.hueLocked ? (
							<button onClick={props.toggleHueLock}>Unlock</button>
						) : (
							<button onClick={props.toggleHueLock}>Lock</button>
						)}
					</div>
				</div>
				<section className="radio-styles">
					<h4>Color schemes:</h4>
					{colorSchemes}
				</section>
				<section className="radio-styles">
					<h4>Color Variations:</h4>
					{colorVariations}
				</section>
				<button className="update-btn">
					<p>Update</p>
				</button>
			</form>
			<footer>
				<button className="close-btn" onClick={props.toggleFullEditBar}>
					Close
				</button>
			</footer>
		</section>
	);
}
