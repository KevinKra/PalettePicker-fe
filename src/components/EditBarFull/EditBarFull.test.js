import React from "react";
import EditBarFull from "./EditBarFull";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

describe("EditBarFull", () => {
  let wrapper;
  const mockUpdateColors = jest.fn();
  beforeEach(() => {
    wrapper = shallow(
      <EditBarFull
        updateColors={mockUpdateColors}
        hueLocked={false}
        editable={false}
      />
    );
  });
  it("should match the snapshot with all data passed in correctly", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it("should updateColors when the form onSubmit event occurs", () => {
    const form = wrapper.find(".edits-form");
    const children = form
      .render()
      .children()
      .children();
    form.simulate("submit", { target: { children } });
    expect(mockUpdateColors).toHaveBeenCalled();
  });
  it("should have transform inline styles if props.editable is true", () => {
    const wrapper = shallow(
      <EditBarFull
        updateColors={mockUpdateColors}
        hueLocked={false}
        editable={true}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it("should render the 'Lock' button when the hueLocked prop is false", () => {
    const lockedButton = wrapper
      .find("button")
      .at(0)
      .text();
    expect(lockedButton).toEqual("Lock");
  });
  it("should render the 'Unlock' button when the hueLocked prop is true", () => {
    const wrapper = shallow(<EditBarFull hueLocked={true} />);
    const unLockedButton = wrapper
      .find("button")
      .at(0)
      .text();
    expect(unLockedButton).toEqual("Unlock");
  });
  it("should invoke the updateColorScheme function when the colorScheme-selection inputs are clicked", () => {
    const updateColorScheme = jest.fn();
    wrapper = shallow(<EditBarFull updateColorScheme={updateColorScheme} />);
    const colorSchemeBtn0 = wrapper
      .find('[name="colorScheme-selection"]')
      .at(0);
    colorSchemeBtn0.simulate("click");
    expect(updateColorScheme).toHaveBeenCalled();
    const colorSchemeBtn1 = wrapper
      .find('[name="colorScheme-selection"]')
      .at(1);
    colorSchemeBtn1.simulate("click");
    expect(updateColorScheme).toHaveBeenCalled();
    const colorSchemeBtn2 = wrapper
      .find('[name="colorScheme-selection"]')
      .at(2);
    colorSchemeBtn2.simulate("click");
    expect(updateColorScheme).toHaveBeenCalled();
    const colorSchemeBtn3 = wrapper
      .find('[name="colorScheme-selection"]')
      .at(3);
    colorSchemeBtn3.simulate("click");
    expect(updateColorScheme).toHaveBeenCalled();
    const colorSchemeBtn4 = wrapper
      .find('[name="colorScheme-selection"]')
      .at(4);
    colorSchemeBtn4.simulate("click");
    expect(updateColorScheme).toHaveBeenCalled();
  });
  it("should invoke the updateVariation function when the variation-selection inputs are clicked", () => {
    const updateVariation = jest.fn();
    wrapper = shallow(<EditBarFull updateVariation={updateVariation} />);
    const variationBtn0 = wrapper.find('[name="variation-selection"]').at(0);
    variationBtn0.simulate("click");
    expect(updateVariation).toHaveBeenCalled();
    const variationBtn1 = wrapper.find('[name="variation-selection"]').at(1);
    variationBtn1.simulate("click");
    expect(updateVariation).toHaveBeenCalled();
    const variationBtn2 = wrapper.find('[name="variation-selection"]').at(2);
    variationBtn2.simulate("click");
    expect(updateVariation).toHaveBeenCalled();
    const variationBtn3 = wrapper.find('[name="variation-selection"]').at(3);
    variationBtn3.simulate("click");
    expect(updateVariation).toHaveBeenCalled();
    const variationBtn4 = wrapper.find('[name="variation-selection"]').at(4);
    variationBtn4.simulate("click");
    expect(updateVariation).toHaveBeenCalled();
    const variationBtn5 = wrapper.find('[name="variation-selection"]').at(5);
    variationBtn5.simulate("click");
    expect(updateVariation).toHaveBeenCalled();
  });
});
