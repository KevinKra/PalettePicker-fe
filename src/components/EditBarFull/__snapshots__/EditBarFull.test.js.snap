// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditBarFull should have transform inline styles if props.editable is true 1`] = `
<section
  className="edit-block"
  style={
    Object {
      "transform": "translateY(-9%)",
    }
  }
>
  <form
    className="edits-form"
    onSubmit={[Function]}
  >
    <div
      className="hue-selection-container"
    >
      <h4>
        Hue Selection:
      </h4>
      <div
        className="hue-selection-inputs"
      >
        <label
          htmlFor="hue-selection"
        >
          <input
            autoComplete="off"
            name="hue-selection"
            placeholder="default: random"
            type="text"
          />
        </label>
        <button>
          Lock
        </button>
      </div>
    </div>
    <section
      className="radio-styles"
    >
      <h4>
        Color schemes:
      </h4>
      <label
        htmlFor="mono"
      >
        mono
        <input
          name="colorScheme-selection"
          onClick={[Function]}
          type="radio"
          value="mono"
        />
      </label>
      <label
        htmlFor="contrast"
      >
        contrast
        <input
          name="colorScheme-selection"
          onClick={[Function]}
          type="radio"
          value="contrast"
        />
      </label>
      <label
        htmlFor="triade"
      >
        triade
        <input
          defaultChecked={true}
          name="colorScheme-selection"
          onClick={[Function]}
          type="radio"
          value="triade"
        />
      </label>
      <label
        htmlFor="tetrade"
      >
        tetrade
        <input
          name="colorScheme-selection"
          onClick={[Function]}
          type="radio"
          value="tetrade"
        />
      </label>
      <label
        htmlFor="analogic"
      >
        analogic
        <input
          name="colorScheme-selection"
          onClick={[Function]}
          type="radio"
          value="analogic"
        />
      </label>
    </section>
    <section
      className="radio-styles"
    >
      <h4>
        Color Variations:
      </h4>
      <label
        htmlFor="default"
      >
        default
        <input
          name="variation-selection"
          onClick={[Function]}
          type="radio"
          value="default"
        />
      </label>
      <label
        htmlFor="pastel"
      >
        pastel
        <input
          defaultChecked={true}
          name="variation-selection"
          onClick={[Function]}
          type="radio"
          value="pastel"
        />
      </label>
      <label
        htmlFor="soft"
      >
        soft
        <input
          name="variation-selection"
          onClick={[Function]}
          type="radio"
          value="soft"
        />
      </label>
      <label
        htmlFor="light"
      >
        light
        <input
          name="variation-selection"
          onClick={[Function]}
          type="radio"
          value="light"
        />
      </label>
      <label
        htmlFor="hard"
      >
        hard
        <input
          name="variation-selection"
          onClick={[Function]}
          type="radio"
          value="hard"
        />
      </label>
      <label
        htmlFor="pale"
      >
        pale
        <input
          name="variation-selection"
          onClick={[Function]}
          type="radio"
          value="pale"
        />
      </label>
    </section>
    <button
      className="update-btn"
    >
      <p>
        Update
      </p>
    </button>
  </form>
</section>
`;

exports[`EditBarFull should match the snapshot with all data passed in correctly 1`] = `
<section
  className="edit-block"
  style={null}
>
  <form
    className="edits-form"
    onSubmit={[Function]}
  >
    <div
      className="hue-selection-container"
    >
      <h4>
        Hue Selection:
      </h4>
      <div
        className="hue-selection-inputs"
      >
        <label
          htmlFor="hue-selection"
        >
          <input
            autoComplete="off"
            name="hue-selection"
            placeholder="default: random"
            type="text"
          />
        </label>
        <button>
          Lock
        </button>
      </div>
    </div>
    <section
      className="radio-styles"
    >
      <h4>
        Color schemes:
      </h4>
      <label
        htmlFor="mono"
      >
        mono
        <input
          name="colorScheme-selection"
          onClick={[Function]}
          type="radio"
          value="mono"
        />
      </label>
      <label
        htmlFor="contrast"
      >
        contrast
        <input
          name="colorScheme-selection"
          onClick={[Function]}
          type="radio"
          value="contrast"
        />
      </label>
      <label
        htmlFor="triade"
      >
        triade
        <input
          defaultChecked={true}
          name="colorScheme-selection"
          onClick={[Function]}
          type="radio"
          value="triade"
        />
      </label>
      <label
        htmlFor="tetrade"
      >
        tetrade
        <input
          name="colorScheme-selection"
          onClick={[Function]}
          type="radio"
          value="tetrade"
        />
      </label>
      <label
        htmlFor="analogic"
      >
        analogic
        <input
          name="colorScheme-selection"
          onClick={[Function]}
          type="radio"
          value="analogic"
        />
      </label>
    </section>
    <section
      className="radio-styles"
    >
      <h4>
        Color Variations:
      </h4>
      <label
        htmlFor="default"
      >
        default
        <input
          name="variation-selection"
          onClick={[Function]}
          type="radio"
          value="default"
        />
      </label>
      <label
        htmlFor="pastel"
      >
        pastel
        <input
          defaultChecked={true}
          name="variation-selection"
          onClick={[Function]}
          type="radio"
          value="pastel"
        />
      </label>
      <label
        htmlFor="soft"
      >
        soft
        <input
          name="variation-selection"
          onClick={[Function]}
          type="radio"
          value="soft"
        />
      </label>
      <label
        htmlFor="light"
      >
        light
        <input
          name="variation-selection"
          onClick={[Function]}
          type="radio"
          value="light"
        />
      </label>
      <label
        htmlFor="hard"
      >
        hard
        <input
          name="variation-selection"
          onClick={[Function]}
          type="radio"
          value="hard"
        />
      </label>
      <label
        htmlFor="pale"
      >
        pale
        <input
          name="variation-selection"
          onClick={[Function]}
          type="radio"
          value="pale"
        />
      </label>
    </section>
    <button
      className="update-btn"
    >
      <p>
        Update
      </p>
    </button>
  </form>
</section>
`;
